modules/core/src/convert.cpp

```diff
@@ -4545,6 +4545,16 @@ static short convertFp16SW(float fp32)
 }
 #endif
 
+#if CV_FP16 && (defined __GNUC__) && (defined __arm__ || defined __aarch64__)
+    #if 5 <= __GNUC__
+    inline float16x4_t load_f16(const short* p) { return vld1_f16((const float16_t*)p); }
+    inline void store_f16(short* p, float16x4_t v) { vst1_f16((float16_t*)p, v); }
+    #else
+    inline float16x4_t load_f16(const short* p) { return (float16x4_t)vld1_s16(p); }
+    inline void store_f16(short* p, float16x4_t v) { vst1_s16(p, (int16x4_t)v); }
```

Comment:
Thanks.  I was wishing to use ```reinterpret_cast_f16_s16``` but as far as I checked, ```reinterpret_cast_f16_s16``` hasn't been implemented same as ```vld1_f16``` in old compilers.
To answer your concern, I confirmed on both
 - Raspberry Pi3 + gcc Raspbian 4.9.2
 - Jetson TK1 + gcc Linaro 4.8.4

and they passed the test.

The alternative was to use the old implementation
```
float16x4_t v_src = *(float16x4_t*)(src+x)
```
but this causes an run time error when the memory address is not aligned.
So, using ```vld_s16``` is kind an alternative of the alternative.

> this branch is used for old compiler only, so no behavior updates are expected. I believe we can use this as-is.

I appreciate your understanding and your review.